### PR TITLE
New: Use Github API Pagination to traverse pages

### LINKED ISSUE #9: Use Github API Pagination to traverse pages
We currently do some custom page traversal across the GitHub API. GitHub asks us not to do that, and makes no guarantees about paging URL structure.

Instead, we should use the paging Links that the GitHub API gives us back in every response:

```js
// console.log(response);
{ data: ...,
  meta: {
    link: '<https://api.github.com/organizations/11639138/repos?page=2&per_page=50&access_token=XXX>; rel="next", <https://api.github.com/organizations/11639138/repos?page=3&per_page=50&access_token=XXX>; rel="last"',
    ...
  }
}
```

### PR DESCRIPTION
**Pull request by stramel**
- [x] CHANGELOG.md has been updated

Fixes #9

Removed the TODO because it was closed with #13

### CODE REVIEW

**Comment by stramel** [comment]
/facepalm.... please don't review yet :sweat:

**Comment by stramel** [comment]
Okay, I think that's better. Leave me some feedback :+1:

File: src/github.ts [hunk 1]
```diff
@@ -188,41 +185,51 @@ export class GitHubConnection {
...
+          .forEach((repo: GitHubRepoData) => {
+            this.setCache(repo.fullName, repo);
+            allRepos.push(repo);
+          });
+
+      if (hasNextPage = !!this._github.hasNextPage(responseMeta)) {
         try {
-          const response = await this._github.repos.getForUser(
-              {username: owner, per_page: pageSize, page: page});
+          const response: any = await this._github.getNextPage(responseMeta);
```
Comment (stramel) [inline src/github.ts]: Wasn't sure why this line specifically was complaining about implicit type any, any thoughts?

**Comment by stramel** [comment]
@FredKSchott Ping :)

File: src/github.ts (hunk 1 above)
Comment (FredKSchott) [reply src/github.ts]: huh, strange. Different TS version?

**Pull request merged**